### ISSUE TITLE
Tensorflow.LookupError : No gradient defined for operation 'cond_7/Rsqrt' (op type: Rsqrt)

### ISSUE THREAD

**Issue by Tomazzzzz** [SciSharp/TensorFlow.NET#830]
Hello,
I am trying to use batch_normalization.
I tried 2 way and got the same error.
First way:
`nn_impl.batch_normalization(y, batchMean, batchVar, beta, gamma, epsilon);`

Second way:
`        protected Tensor BatchNorm(Tensor x, Tensor gamma, Tensor beta, Tensor movingMean, Tensor movingVar, float eps)
        {
            var inv = tf.cast(math_ops.rsqrt(movingVar + eps), x.dtype);
            inv *= gamma;
            return x * inv + (beta - movingMean * inv);
        }`

THNX!!!

**Comment by Tomazzzzz** [comment]
Hello, 
Another issue with gradient :(
Tensorflow.LookupError : No gradient defined for operation 'Elu' (op type: Elu)

**Comment by UCtreespring** [comment]
Sovled?
batch_normalization  trouble -----( version  Tensorflow.NET 0.40.1 Tensorflow.Keras 0.5.1)
I need your help.
like some example code about using BN function in NeuralNetwork Code.
Thanks.

**Comment by wx92nlp** [comment]
same thing
i used nn_impl.batch_normalization
but :No gradient

**Comment by Tomazzzzz** [comment]
Hello,
I have used function where I manually register gradient.

```
            ops.RegisterGradientFunction("Elu", (oper, out_grads) =>
            {
                var y = oper.outputs[0];
                return new[] { gen_ops.elu_grad(out_grads[0], y) };
            });
        
```

**Comment by UCtreespring** [comment]
> 
Thank you so much!

------------------------------
A custom gradient function？
Cool 
But It doesn't sound easy for me. what a pity. 
I'm Freshmen, I don't even know where this code should be.

**Comment by UCtreespring** [comment]
> Hello, I have used function where I manually register gradient.
> 
> ```
>             ops.RegisterGradientFunction("Elu", (oper, out_grads) =>
>             {
>                 var y = oper.outputs[0];
>                 return new[] { gen_ops.elu_grad(out_grads[0], y) };
>             });
>         
> ```

Thank you so much!
__________________________________________________________________

A custom gradient function？
Cool
But It doesn't sound easy for me. what a pity.
I'm Freshmen, I don't even know where this code should be.

**Comment by Tomazzzzz** [comment]
@UCtreespring Hi, just call it in constructor before creating/loading graph.
